python/ray/data/tests/test_iterator.py

```diff
@@ -158,6 +171,18 @@ def test_torch_conversion(ray_start_regular_shared):
     ), iter_batches_calls_kwargs
 
 
+def test_torch_multi_use_iterator(ray_start_regular_shared):
+    """Tests that the iterator outputted by `iter_torch_batches` can be used
+    multiple times."""
+    ds = ray.data.range(5)
+    it = ds.iterator().iter_torch_batches()
+
+    for _ in range(2):
+        for batch in it:
+            assert isinstance(batch["id"], torch.Tensor)
+            assert batch["id"].tolist() == list(range(5))
```

Comment:
could we also test the case that iteration is stopped in the middle, without consuming all batches? This is to simulate PyTorch lightning behavior.

```
for _ in range(2):
  for i, batch in enumerate(it):
      assert ...
      if i > N:
         break
```